src/openrct2-ui/CMakeLists.txt

```diff
@@ -157,7 +157,7 @@ endif ()
 # Only valid for Clang for now:
 # - GCC 8 does not support -Wno-pragma-once-outside-header
 # - Other compilers status unknown
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+if (ENABLE_HEADERS_CHECK AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
```

Comment:
We could just switch to using a different way of handling the header checker then it would be compatible on all compilers (i.e. do it the openloco way create a cpp for every header that only includes the header)

Reply:
I'm not in favour of this, in general I'm not a fan of creating files when configuring, that slows things down especially on windows and confuses people new to the project by introducing errors in files that are not strictly part of the repo. I'd rather stay with current approach even if we can only use clang - I think it's enough for our needs.